dimitriosv/week4/app/components/Messages.jsx

```diff
@@ -0,0 +1,50 @@
+//import Message from './Message.jsx';
+var Message = require('./Message');
+var React = require('react');
+var ReactDOM = require('react-dom');
+
+
+var Messages = React.createClass({
+    render: function () {
+        
+        return  <div>
+                    <div>
+                        {this.props.messages
+                            .filter(function(message){ 
+                                return !message.isArchived;
+                            })
+                            .map(function(message){
+                                return (<div className="message" key={message.messageId}  >
+                                        <Message  messageId={message.messageId} 
+                                            messageText={message.text}
+                                            messageIsArchived={message.isArchived} 
+                                            deleteMessageParent={this.props.deleteMessageParent}
+                                            toggleArchiveMessageParent={this.props.toggleArchiveMessageParent}
+                                            allLists={this.props.allLists}
+                                        />
+                                    </div>)
+                            }.bind(this))}
+                    </div>
+                    <div>Archived:</div>
+                    <div>
+                        {this.props.messages
+                            .filter(function(message){ 
+                                return message.isArchived;
+                            })
+                            .map(function(message){
+                                return (<div className="message" key={message.messageId}  >
+                                        <Message  messageId={message.messageId} 
```

Comment:
looks to me that you could just pass in the `message` object rather than spreading all the props into it.